cmake/Futility_Configurations.cmake

```diff
@@ -191,7 +191,7 @@ IF(CMAKE_Fortran_COMPILER_ID STREQUAL "Intel" OR
 
     SET(Fortran_FLAGS_RELEASE
         ${CSYM}O3
-        ${CSYM}Os
+        ${CSYM}march=native
```

Comment:
Not familiar with these flags... what did the old one do, and what does the new one do?

Reply:
additionally, these are changes to Intel compiler flags...was Intel actually tested?

Reply:
The intel flag should have been dropped; I'll correct this
-march=native enables cpu-specific instructions; such as avx512. These are not enabled by default, even when using O3. When enabling this option, I saw runtimes for 5a-2d drop from 10 minutes to 7.5 for standard PETSc runs.

Reply:
I've been reading a bit about this option.  I THINK it is safe to use, and that speedup is certainly attractive.  However, I think it would be good to do some more extensive testing before we merge this in, such as:
1. Run heavies with and without the flag to ensure there are no performance degradations across a larger set of problems
2. Ensure this isn't going to cause any issues for clusters.  Reading about the flag a bit just now, I found the following:
![image](https://user-images.githubusercontent.com/9356636/75553505-81128680-5a06-11ea-9cd4-c06eae345f1b.png)
My concern in that case is what if someone builds on a head node of a cluster and then runs on the compute nodes?  My expectation is that the CPU between the two will generally be the same or very similar and it's probably fine, but I think it'd be good to test some builds on a couple different machines before putting this on master.

Reply:
An additional thought: we've talked a bit about supporting more compilers, so IF we add this as a default for gfortran, let's leave it enabled for Intel as well.

Reply:
I had done a good bit of testing using machine specific optimizations with MPACT. For a single assembly ge14 case with TH turned on I saw ~2% speedup (it was consistent, i.e., not likely to be just noise). However, many many tests failed when doing a full ctest. The reason was due to rounding differences making some .APPROXEQA. to fail. I saw that setting -mfpmath=sse also helps slightly.

Reply:
@mattkab2 Need to do a full MPACT build using this Futility branch to see if there are any failures introduced by this compiler flag change.

Reply:
It seems my comments were lost or not posting for some reason. Here's a summary:
These command line changes have a strong chance of breaking things when the processors on login nodes are newer than the compute nodes, even by a single model. @djabaay and I actually had a conversation about this some time ago and one suggestion that came up was to perhaps add another build type "optimized" for developers or cluster admins that creates a non-portable executable.
@aarograph Yes, I already have one actually, and I've identified and fixed the floating point errors that occur when enabling march=native. I've been iterating between that branch and these changes to try and minimize the disruption to code. Since some changes on the MPACT side are required to integrate GMRES completely anyways, one option I'm leaning towards is to merge this code in without the command line changes, update on the MPACT side, then add this back in afterwards.

Reply:
@mattkab2 @djabaay There are a lot of potentially issues with changing release compiler flags.  If we end up doing this, it needs to be done completely separately from any other changes so that it can be thoroughly vetted and easily reverted if necessary.

Back out this change and any downstream MPACT changes you've made that were due to this compiler flag change.  If you'd like to keep them, move them to a separate branch, but don't merge them in with any of this GMRES work.

Reply:
Sounds good @aarograh